### ISSUE TITLE
Add an API to reach OpenMP threads

### ISSUE THREAD

**Issue by vallsv** [silx-kit/silx#1781]
There is no easy way to know if the library was linked with OpenMP. Cython do not provide abstraction. This information is important if we want to custom the algorithm we want to use.

OpenMP provides this API:
```
int omp_get_max_threads()
int omp_get_num_procs()
int omp_get_num_threads()
```
At least we can bind this API. Maybe on `silx.utils.openmp`. If OpenMP was not used it can returns 1 by default.

**Comment by t20100** [comment]
Yes it would be nice to get information about how silx was built, but maybe it is separated from openmp though (something like Python sysconfig).

**Comment by kif** [comment]
On Mon, 23 Apr 2018 08:00:41 +0000 (UTC)
Thomas VINCENT <notifications@github.com> wrote:

> Yes it would be nice to get information about how silx was built, but maybe it is separated from openmp though (something like Python sysconfig).

+1

-- 
Jérôme Kieffer
tel +33 476 882 445

**Comment by vallsv** [comment]
We do not have that now, but only a way to to know if it was compiled using OpenMP.
It is enough right now to remove it from the 0.8 milestones.

**Comment by payno** [comment]
Even if the feature seems nice, this is a 8 years old issue. I would say at the moment this is a won't have. Doesn't seems like a priority.

If it is please feel free to reopen with some arguments / PR of implementation.

**Comment by kif** [comment]
The comment of @t20100 remains valid: was silx compiled with OpenMP activated ?
I have to admit that with meson, it is less of a problem.

**Comment by t20100** [comment]
> I have to admit that with meson, it is less of a problem.

@kif could you give more insight? any way with meson to gather/store the build option and make it available from python?


It turns out it is still required, see #4650 
Let's discuss this in  #4650
